src/main/java/uk/gov/digital/ho/hocs/client/audit/AuditClient.java

```diff
@@ -85,22 +83,28 @@ private void sendAuditMessage(EventType eventType, UUID caseUUID, UUID stageUUID
                 clientContext.getUserId());
 
         try {
-            Map<String, Object> queueHeaders = getQueueHeaders(eventType.toString());
             String jsonPayload = objectMapper.writeValueAsString(request);
-            producerTemplate.sendBodyAndHeaders(auditTopic, jsonPayload, queueHeaders);
+
+            var publishRequest = new PublishRequest()
+                    .withTopicArn(snsProperties.getAudit().getArn())
+                    .withMessage(jsonPayload)
+                    .withMessageAttributes(getQueueHeaders(eventType.toString()));
+
+            snsClient.publish(publishRequest);
+
             log.info("Create audit of type {} for Case UUID: {}, correlationID: {}, UserID: {}, event: {}",
                     eventType, caseUUID, clientContext.getCorrelationId(), clientContext.getUserId(), value(EVENT, AUDIT_EVENT_CREATED));
         } catch (Exception e) {
             log.error("Failed to create audit event for case UUID {}, event {}, exception: {}", caseUUID, value(EVENT, AUDIT_FAILED), value(EXCEPTION, e));
         }
     }
 
-    private Map<String, Object> getQueueHeaders(String eventType) {
+    private Map<String, MessageAttributeValue> getQueueHeaders(String eventType) {
         return Map.of(
-                EVENT_TYPE_HEADER, eventType,
-                ClientContext.CORRELATION_ID_HEADER, clientContext.getCorrelationId(),
-                ClientContext.USER_ID_HEADER, clientContext.getUserId(),
-                ClientContext.GROUP_HEADER, clientContext.getGroups());
+                EVENT_TYPE_HEADER, new MessageAttributeValue().withDataType("String").withStringValue(eventType),
```

Comment:
MessageAttributeValue can have a public enum to represent the dataType. Not good to keep passing a literal
If messageAttributeValue cannot work without both parameters being provided then consider a simple contractor instead with validation.
In either case, it will need validation as a builder without validation is much the same as a Lombok annotated mojo!

Reply:
I have added a `StringMessageAttributeValue` wrapper class that sets the type for us. I have also added a not-empty validator to the value.